reader_test.go

```diff
@@ -1669,3 +1671,159 @@ func TestErrorCannotConnectGroupSubscription(t *testing.T) {
 			"must fail when it cannot connect")
 	}
 }
+
+// Tests that the reader can read zero message batches from log compacted
+// topics.
+//
+// This test forces varying sized chunks of duplicated messages along with
+// configuring the topic with a minimal `segment.bytes` in order to
+// guarantee that at least 1 batch can be compacted down to 0 messages.
```

Comment:
It seems to me that there cannot be batch without messages. Our case here is that can be batch without **_new_** (unread by current conn) messages at the end of the batch. If kafka's segment was compacted down to 0 messages, it would be skipped during Fetch and kafka would try to return batch from next segment.

Test case here leads to segment been compacted to at least 1 message, but not at the end of segment. I tried to create fully compacted segments and they were skipped by kafka during fetch..

Reply:
thanks for the clarification, made some updates to the comments